flow-server/src/main/java/com/vaadin/flow/di/DefaultInstantiator.java

```
@@ -129,24 +148,52 @@ private I18NProvider getI18NProviderInstance() {
         return null;
     }
 
+    private MenuAccessControl getMenuAccessControlInstance() {
+        String property = getInitProperty(InitParameters.MENU_ACCESS_CONTROL);
+        if (property == null) {
+            return new DefaultMenuAccessControl();
+        }
+        try {
+            // Get Menu Access Control class if found in application
+            // properties
+            Class<?> providerClass = DefaultInstantiator.class.getClassLoader()
+                    .loadClass(property);
+            if (MenuAccessControl.class.isAssignableFrom(providerClass)) {
+
+                return ReflectTools.createInstance(
+                        (Class<? extends MenuAccessControl>) providerClass);
+            } else {
+                LoggerFactory.getLogger(DefaultInstantiator.class).warn(
+                        "Menu access control implementation class property '{}' is set to '{}' but it's not {} implementation",
+                        InitParameters.MENU_ACCESS_CONTROL, property,
+                        MenuAccessControl.class.getSimpleName());
```

Looking at the implementation, in this case the `MenuAccessControl` feature is disabled, since the method will return `null`.
I wonder if it would be better to throw an exception, since it is definitely a configuration error.
Otherwise, I would expand the log message to explicitly say that `MenuAccessControl` feature will be disabled.

---

Let's throw an exception of InvalidIMenuAccessControlException type.